packages/runtime/src/utils.ts

```diff
@@ -0,0 +1,19 @@
+export type Dependency = number | string | { uniqueId: string; }
+
+export function isDependency(element: any): element is Dependency {
+  return typeof element === 'number' || typeof element === 'string' || (typeof element === 'object' && element.uniqueId !== undefined);
+}
+
+export const calculateCacheKey = (modulePath: string, dependencies: Dependency[]): string => {
```

Comment:
It makes the cache key depend on the file's relative path to the Current Working Directory (CWD) instead of the relative path to the current file. This will help to cache the file value if it's called in different files. I needed to use it with Popmenu because the `Announcements Query` is not called in any component that is server-side rendered, so I needed to call it on the server side inside another component to cache it.